src/utils/api.js

```diff
@@ -14,7 +14,16 @@ async function fetchApiCall({ url, method, body = null, successCallback = null,
 
   try {
     const response = await fetch(url, config)
+
+    // Check for empty or non-JSON response
+    if (response.status === 204 || response.status === 301) {
```

Comment:
Hmm, this seems oddly specific. Why do we need to handle these specific codes this way? Is there a better way than checking status codes?

Reply:
This seems like some boilerplate/copied code... I don't think we will get either of those codes from our server response

Reply:
`[19/Jan/2024 00:09:27] "DELETE /api/datasessions/23 HTTP/1.1" 301 0
[19/Jan/2024 00:09:27] "DELETE /api/datasessions/23/ HTTP/1.1" 204 0`

The api sends these two responses from the server on a Delete request. Both which have empty responses that the fetchAPI method can't handle currently. But, I think a better way would be checking the response body itself for an empty one and handling that instead of the specific codes. Before it would send an error whenever I sent a DELETE